npm/pkg/dataplane/dataplane.go

```diff
@@ -451,3 +547,7 @@ func (dp *DataPlane) deleteIPSetsAndReferences(sets []*ipsets.TranslatedIPSet, n
 	}
 	return nil
 }
+
+func (dp *DataPlane) configuredToApplyInBackground() bool {
+	return util.IsWindowsDP() && dp.ApplyInBackground
```

Comment:
Instead of checking for isWindowsDP everytime we need to applyDP, can we assign a global DP variable which we can set once in INIT and then only keep checking that value ?